lang/cs/Org.Apache.REEF.Driver/Bridge/ClrBridge.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using Org.Apache.REEF.Bridge;
+using org.apache.reef.bridge.message;
+using Org.Apache.REEF.Wake.Avro;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.Wake.Remote;
+using Org.Apache.REEF.Tang.Annotations;
+using System.Threading;
+
+namespace Org.Apache.REEF.Driver.Bridge
+{
+    /// <summary>
+    /// An Observer implementation which handles all of the messages defined in
+    /// the Java to C# Avro protocol coming from the Java bridge which invokes
+    /// the appropriate target method in the C# side of the driver.
+    /// </summary>
+    internal sealed class ClrBridge : IObserver<IMessageInstance<SystemOnStart>>
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(ClrBridge));
+        private static long identifierSource = 0;
```

Comment:
do we need to synchronize it?

Reply:
There is an Interlocked increment on line 66. In C# the variables are not declared atomic like in Java or C#, only the operation is interlocked. See example on this page: https://msdn.microsoft.com/en-us/library/system.threading.interlocked(v=vs.110).aspx

Reply:
oh, right. please ignore :)